tests/system/src/test/java/test/javafx/stage/SystemMenuBarTest.java

```diff
@@ -108,4 +115,64 @@ public MenuBar createFailingMenuBar() {
 
         return menuBar;
     }
+
+    @Test
+    public void testMemoryLeak() throws InterruptedException {
+        Util.runAndWait(() -> {
+            Thread.currentThread().setUncaughtExceptionHandler((t,e) -> {
+                e.printStackTrace();
+                failed.set(true);
+                memoryLatch.countDown();
+            });
+            createMenuBarWithItemsStage();
+        });
+        memoryLatch.await();
+        assertFalse(failed.get());
+    }
+
+    public void createMenuBarWithItemsStage() {
+        final ArrayList<WeakReference<MenuItem>> uncollectedMenuItems = new ArrayList<>();
+
+        Stage stage = new Stage();
+        VBox root = new VBox();
+        final MenuBar menuBar = new MenuBar();
+        final Menu menu = new Menu("MyMenu");
+        menuBar.getMenus().add(menu);
+        menuBar.setUseSystemMenuBar(true);
+        root.getChildren().add(menuBar);
+
+        Scene scene = new Scene(root);
+        stage.setScene(scene);
+        stage.show();
+        stage.requestFocus();
+        Thread t = new Thread(){
+            @Override public void run() {
+                for (int i = 0; i < 10; i++) {
+                    try {
+                        Thread.sleep(20);
+                    } catch (Exception e) {
+                        e.printStackTrace();
+                    }
+                    Platform.runLater(() -> {
+                        menu.getItems().clear();
+                        MenuItem menuItem = new MenuItem("MyItem");
+                        WeakReference wr = new WeakReference<>(menuItem);
+                        uncollectedMenuItems.add(wr);
+                        menu.getItems().add(menuItem);
+                    });
+                }
+                Platform.runLater( () -> {
```

Comment:
No, it is needed because we want to give the JavaFX platform the opportunity to do "whatever is needed" after the last runLater() is done. Hence, we put what we want to do on the Runnable queue of the FX thread, so that we don't depend on internals.

Reply:
This makes sense. To answer one of Jose's questions:

> Is it guaranteed that it will be called only after the previous one ends?

Yes, this is specified behavior of `Platform::runLater`.